src/app/(customerFacing)/products/page.tsx

```diff
@@ -40,21 +56,27 @@ export default function ProductsPage({
               </>
             }
           >
-            <ProductsSuspense />
+            <ProductsSuspense options={options} />
           </Suspense>
         </div>
       </section>
     </div>
   );
 }
 
-async function ProductsSuspense() {
-  const products = await getAllProducts();
+async function ProductsSuspense({ options }: { options: { sortField: string, sortOrder: "asc" | "desc" | "new" } }) {
+
+  if(options.sortOrder == 'new'){
+    options.sortField = 'createdAt'
+    options.sortOrder = 'desc'
+  }
```

Comment:
in `getSortOptions` you can say
```suggestion
  const sortOrder = sortValue === "new" ? "desc" : sortValue || "desc";
```
This way you'd avoid the if statement here